packages/roosterjs-editor-dom/lib/utils/setListItemStyle.ts

```diff
@@ -0,0 +1,56 @@
+import ContentTraverser from '../contentTraverser/ContentTraverser';
+import getStyles from '../style/getStyles';
+import safeInstanceOf from './safeInstanceOf';
+import setStyles from '../style/setStyles';
+import { InlineElement } from 'roosterjs-editor-types';
+
+/**
+ * Set the Style of a List Item provided, with the styles that the inline child elements have
+ * If the child inline elements have different styles, it will not modify the styles of the list item
+ * @param element the LI Element to set the styles
+ * @param styles The styles that should be applied to the element.
+ */
+export function setListItemStyle(element: HTMLLIElement, styles: string[]) {
+    const elementsStyles = getInlineChildElementsStyle(element);
+    let stylesToApply: Record<string, string> = getStyles(element);
+
+    styles.forEach(styleName => {
+        const styleValues = elementsStyles.map(style =>
+            style[styleName] !== undefined ? style[styleName] : ''
+        );
+
+        if (
+            styleValues &&
+            (styleValues.length == 1 || new Set(styleValues).size == 1) &&
+            styleValues[0]
+        ) {
+            stylesToApply[styleName] = styleValues[0];
+        }
+    });
+    setStyles(element, stylesToApply);
+}
+function getInlineChildElementsStyle(element: HTMLElement) {
+    const result: Record<string, string>[] = [];
+    const contentTraverser = ContentTraverser.createBodyTraverser(element);
+    let currentInlineElement: InlineElement;
+
+    while (contentTraverser.currentInlineElement != currentInlineElement) {
```

Comment:
Interesting, I thought when go beyond the boundary it currentInlineElement should return null.
Maybe it is a bug?

Reply:
Yes, maybe something is up there. 
If the currentInlineElement was the last element, the method getNextInlineElement was returning the last element instead of null sometimes.

Reply:
Please open an issue to track. We need to revisit all places where this method is used.